### PR TITLE
Removed the legacy session based runtime

### PR DESCRIPTION
**Pull request by mhong**
, in preparation to extend the eager based runtime with tracing support.

### CODE REVIEW

**Comment by mhong** [comment]
@swift-ci please test tensorflow

**Comment by mhong** [comment]
@swift-ci please test tensorflow

**Comment by mhong** [comment]
@swift-ci please test tensorflow

File: stdlib/public/TensorFlow/CompilerRuntime.swift [hunk 1]
```diff
@@ -856,10 +620,7 @@ public final class _TensorComputation {
   // TODO(hongm): Retire returnValues when eager based runtime is removed.
   var returnValues: [CTensorHandle?]
 
-  // NOTE: the following properties are intentionally not implemented as an enum
-  // due to high churn, *please do not refactor for Swiftiness*.
-  private var stateTFE: TFEState?
-  private var stateTF: TFState?
+  private var stateTFE: TFEState
```
Comment (rxwei) [inline stdlib/public/TensorFlow/CompilerRuntime.swift]: If that's the only state, how about just `state`?
```suggestion
  private var state: TFEState
```

File: stdlib/public/TensorFlow/CompilerRuntime.swift [hunk 2]
```diff
@@ -254,13 +238,8 @@ public final class _ExecutionContext {
   // NOTE: the following properties are intentionally not implemented as an enum
   // due to high churn, *please do not refactor for Swiftiness*.
   /// The set of all loaded programs indexed by their unique address.
-  /// Used when _RuntimeConfig.usesTFEagerAPI is true.
   private var loadedTFEPrograms: [UnsafeRawPointer : CTFGraph] = [:]
```
Comment (rxwei) [inline stdlib/public/TensorFlow/CompilerRuntime.swift]: Seems `loadedPrograms` is better.

**Review by rxwei — APPROVED** [review]

**Review by bgogul — APPROVED** [review]
Nice!

File: stdlib/public/TensorFlow/CompilerRuntime.swift (hunk 1 above)
Comment (mhong) [reply stdlib/public/TensorFlow/CompilerRuntime.swift]: done

File: stdlib/public/TensorFlow/CompilerRuntime.swift (hunk 2 above)
Comment (mhong) [reply stdlib/public/TensorFlow/CompilerRuntime.swift]: done

**Comment by mhong** [comment]
Thanks for the review!

**Pull request merged**